Assets/Editor/WeaponDrawer.cs

```diff
@@ -0,0 +1,49 @@
+using UnityEditor;
+using UnityEngine.UIElements;
+using UnityEditor.UIElements;
+using UnityEngine;
+
+[CustomPropertyDrawer(typeof(Weapon))]
+public class WeaponDrawer : PropertyDrawer
+{
+    public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
```

Comment:
Please remember to rework this and delete `CreatePropertyGUI` if it still isn't in use.

Reply:
As stated in a PM, this could be removed since Unity does have editor GUI for structs by default :upside_down_face: